autoPyTorch/pipeline/components/setup/network/backbone/image.py

```diff
@@ -0,0 +1,229 @@
+                                bn_size=bn_size, growth_rate=growth_rate, drop_rate=drop_rate)
+            features.add_module('denseblock%d' % (i + 1), block)
+            num_features = num_features + num_layers * growth_rate
+            if i != len(block_config) - 1:
+                trans = _Transition(num_input_features=num_features, num_output_features=num_features // 2,
+                                    pool_size=2 if i > len(block_config) - division_steps else 1)
+                features.add_module('transition%d' % (i + 1), trans)
+                num_features = num_features // 2
+
+        # Final batch norm
+        features.add_module('last_norm', nn.BatchNorm2d(num_features))
+        self.backbone = features
+        return features
+
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, str]] = None) -> Dict[str, Any]:
+        return {
+            'shortname': 'DenseNetBackbone',
+            'name': 'DenseNetBackbone',
+        }
+
+    @staticmethod
+    def get_hyperparameter_search_space(dataset_properties: Optional[Dict[str, str]] = None,
+                                        min_growth_rate: int = 12,
+                                        max_growth_rate: int = 40,
+                                        min_num_blocks: int = 3,
+                                        max_num_blocks: int = 4,
+                                        min_num_layers: int = 4,
+                                        max_num_layers: int = 64) -> ConfigurationSpace:
+        cs = CS.ConfigurationSpace()
+        growth_rate_hp = UniformIntegerHyperparameter('growth_rate',
+                                                      lower=min_growth_rate,
+                                                      upper=max_growth_rate)
+        cs.add_hyperparameter(growth_rate_hp)
+
+        blocks_hp = UniformIntegerHyperparameter('blocks',
+                                                 lower=min_num_blocks,
+                                                 upper=max_num_blocks)
+        cs.add_hyperparameter(blocks_hp)
+        use_dropout = CategoricalHyperparameter('use_dropout', choices=[True, False])
+        dropout = UniformFloatHyperparameter('dropout',
+                                             lower=0.0,
+                                             upper=1.0)
+        cs.add_hyperparameters([use_dropout, dropout])
+        cs.add_condition(CS.EqualsCondition(dropout, use_dropout, True))
+
+        for i in range(1, max_num_blocks + 1):
+            layer_hp = UniformIntegerHyperparameter('layer_in_block_%d' % i,
+                                                    lower=min_num_layers,
+                                                    upper=max_num_layers)
+            cs.add_hyperparameter(layer_hp)
+
+            if i > min_num_blocks:
+                cs.add_condition(CS.GreaterThanCondition(layer_hp, blocks_hp, i - 1))
+
+        return cs
```

Comment:
We definitely want a wide resnet for image data with shake drop and shake shake regularizations as options (see https://github.com/automl/Auto-PyTorch/blob/master/autoPyTorch/components/networks/image/resnet.py). Does not have to be now, but if we don't add it now please create an issue so that we don't forget it

Reply:
see #32